File: src/main/java/com/ggardet/doctor/spi/dto/QualificationDTO.java
package com.ggardet.doctor.spi.dto;

import jakarta.validation.constraints.NotBlank;

import java.time.LocalDate;
import java.util.UUID;

public record QualificationDTO(
        UUID id,

        @NotBlank String title,

        String institution,

        LocalDate graduationDate
) {
}

File: src/main/kotlin/com/ggardet/doctor/spi/dto/QualificationDTO.kt
package com.ggardet.doctor.spi.dto

import jakarta.validation.constraints.NotBlank
import java.time.LocalDate
import java.util.*

data class QualificationDTO(
    val id: UUID?,
    
    @field:NotBlank
    val title: String,
    
    val institution: String?,
    
    val graduationDate: LocalDate?
)